src/mito2/src/memtable/time_series.rs

```diff
@@ -54,7 +54,7 @@ use crate::region::options::MergeMode;
 use crate::row_converter::{DensePrimaryKeyCodec, PrimaryKeyCodecExt};
 
 /// Initial vector builder capacity.
-const INITIAL_BUILDER_CAPACITY: usize = 0;
+const BUILDER_CAPACITY: usize = 512;
```

Comment:
The init capacity 0 is to avoid each Series pre-allocating too much memory when the points in each series are far less than the capacity.

For example, if we have 8k series and each series has 100 points, then the memtable may use 5x memory than before. What's more, the write buffer size doesn't consider the builder capacity now, so we may need to bump the write buffer size once we create a builder.

Maybe we can freeze the builder earlier once the number of items reaches a threshold.

Reply:
Add a two-step capacity that starts from 16